pkg/apiserver/rest/usecase/definition.go

```diff
@@ -108,23 +116,184 @@ func (d *definitionUsecaseImpl) listDefinitions(ctx context.Context, list *unstr
+	var customSchemaMap = make(map[string]*utils.UIParameter, len(customSchema))
+	for i, custom := range customSchema {
+		customSchemaMap[custom.JSONKey] = customSchema[i]
+	}
+	for i := range defaultSchema {
+		dSchema := defaultSchema[i]
+		if cusSchema, exist := customSchemaMap[dSchema.JSONKey]; exist {
+			if cusSchema.Description != "" {
+				dSchema.Description = cusSchema.Description
+			}
+			if cusSchema.Label != "" {
+				dSchema.Label = cusSchema.Label
+			}
+			if cusSchema.SubParameterGroupOption != nil {
+				dSchema.SubParameterGroupOption = cusSchema.SubParameterGroupOption
+			}
+			if cusSchema.Validate != nil {
+				dSchema.Validate = cusSchema.Validate
+			}
+			if cusSchema.UIType != "" {
+				dSchema.UIType = cusSchema.UIType
+			}
+			if cusSchema.Disable != nil {
+				dSchema.Disable = cusSchema.Disable
+			}
+			if cusSchema.SubParameters != nil {
+				dSchema.SubParameters = patchSchema(dSchema.SubParameters, cusSchema.SubParameters)
+			}
+			if cusSchema.Sort != 0 {
+				dSchema.Sort = cusSchema.Sort
+			}
+		}
+	}
+	sort.Slice(defaultSchema, func(i, j int) bool {
+		return defaultSchema[i].Sort < defaultSchema[j].Sort
+	})
+	return defaultSchema
+}
+
+func renderDefaultUISchema(apiSchema *openapi3.Schema) []*utils.UIParameter {
+	if apiSchema == nil {
+		return nil
+	}
+	var params []*utils.UIParameter
+	for key, property := range apiSchema.Properties {
+		if property.Value != nil {
+			param := renderUIParameter(key, utils.FirstUpper(key), property, apiSchema.Required)
+			params = append(params, param)
+		}
+	}
+	return params
+}
+
+func renderUIParameter(key, label string, property *openapi3.SchemaRef, required []string) *utils.UIParameter {
+	var parameter utils.UIParameter
+	subType := ""
+	if property.Value.Items != nil {
+		if property.Value.Items.Value != nil {
+			subType = property.Value.Items.Value.Type
+		}
+		parameter.SubParameters = renderDefaultUISchema(property.Value.Items.Value)
+	}
+	if property.Value.Properties != nil {
+		parameter.SubParameters = renderDefaultUISchema(property.Value)
+	}
+	parameter.Validate = &utils.Validate{}
+	parameter.Validate.DefaultValue = property.Value.Default
+	for _, enum := range property.Value.Enum {
+		parameter.Validate.Options = append(parameter.Validate.Options, utils.Option{Label: utils.RenderLabel(enum), Value: enum})
+	}
+	parameter.JSONKey = key
+	parameter.Description = property.Value.Description
+	parameter.Label = label
+	parameter.UIType = utils.GetDefaultUIType(property.Value.Type, len(parameter.Validate.Options) != 0, subType)
+	parameter.Validate.Max = property.Value.Max
+	parameter.Validate.MaxLength = property.Value.MaxLength
+	parameter.Validate.Min = property.Value.Min
+	parameter.Validate.MinLength = property.Value.MinLength
+	parameter.Validate.Pattern = property.Value.Pattern
+	parameter.Validate.Required = utils.StringsContain(required, property.Value.Title)
+	parameter.Sort = 100
```

Comment:
What does this parameter mean?

Reply:
Corresponds to a UI component on the front end, which is a form in most cases.

Reply:
What does `Sort` mean?

Reply:
Order of precedence